packages/commonwealth/client/scripts/views/components/component_kit/cw_socials.tsx

```diff
@@ -12,97 +11,97 @@ import { CWIconButton } from './cw_icon_button';
 import type { IconName } from './cw_icons/cw_icon_lookup';
 import { CWIcon } from './cw_icons/cw_icon';
 
-type SocialsAttrs = {
+type SocialsProps = {
   socials: string[];
   handleInputChange?: (value: string[]) => void;
 };
 
-export class CWSocials extends ClassComponent<SocialsAttrs> {
-  private socials: string[];
+export const CWSocials = (props: SocialsProps) => {
+  const [socials, setSocials] = useState([]);
 
-  private addInputRow = () => {
-    this.socials = [...this.socials, ''];
+  const addInputRow = () => {
+    setSocials([...socials, '']);
   };
 
-  private deleteInputRow = (
-    index: number,
-    handleInputChange: (value: string[]) => void
-  ) => {
-    this.socials = this.socials.filter((_, i) => i !== index);
-    handleInputChange(this.socials);
+  const deleteInputRow = (index: number) => {
+    setSocials(socials.filter((_, i) => i !== index));
   };
 
-  oninit(vnode: ResultNode<SocialsAttrs>) {
-    this.socials = vnode.attrs.socials ? [...vnode.attrs.socials] : [];
-  }
+  useEffect(() => {
+    if (props.socials) {
+      setSocials([...props.socials]);
+    }
+  }, []);
```

Comment:
this is not the way how we should initialize things in react. 
just do   const [socials, setSocials] = useState(props.socials);